app/controllers/simplified-account/settings/team-members/team-members.controller.test.js

```diff
@@ -1,104 +1,88 @@
-    {
-      external_id: 'user-id-for-view-only-user',
-      service_roles: {
-        service:
-            {
-              service: { external_id: SERVICE_ID },
-              role: { name: 'view-only' }
-            }
-      }
-    }))
-  const users = [adminUser, viewOnlyUser]
-  getServiceUsersStub = sinon.stub().resolves(users)
+const users = [adminUser, viewOnlyUser]
+const invitedUsers = [
+  { email: 'invited-admin-user@user.gov.uk', role: 'admin' },
+  { email: 'invited-view-only-user@user.gov.uk', role: 'view-only' }
+]
 
-  const invitedAdminUser = { email: 'invited-admin-user@user.gov.uk', role: 'admin' }
-  const invitedViewOnlyUser = { email: 'invited-view-only-user@user.gov.uk', role: 'view-only' }
-  const invitedUsers = [invitedAdminUser, invitedViewOnlyUser]
-  getInvitedUsersStub = sinon.stub().resolves(invitedUsers)
+const mockResponse = sinon.spy()
+const mockGetServiceUsers = sinon.stub().resolves(users)
+const mockGetInvitedUsers = sinon.stub().resolves(invitedUsers)
+
+const { res, nextRequest, call } = new ControllerTestBuilder('@controllers/simplified-account/settings/team-members/team-members.controller')
+  .withServiceExternalId(SERVICE_ID)
+  .withAccountType(ACCOUNT_TYPE)
+  .withStubs({
+    '@utils/response': { response: mockResponse },
+    '@services/user.service':
+      { getServiceUsers: mockGetServiceUsers, getInvitedUsers: mockGetInvitedUsers }
 
-  teamMembersController = getController({
-    response: responseStub,
-    getServiceUsers: getServiceUsersStub,
-    getInvitedUsers: getInvitedUsersStub,
-    ...additionalStubs
   })
-  res = {
-    redirect: sinon.spy()
-  }
-  req = {
-    user: adminUser,
-    service: {
-      externalId: SERVICE_ID
-    },
-    account: {
-      type: ACCOUNT_TYPE
-    },
-    ...additionalReqProps
-  }
-  teamMembersController[method](req, res)
-}
+  .build()
 
 describe('Controller: settings/team-members', () => {
   describe('get', () => {
-    before(() => setupTest('get'))
+    before(() => {
+      nextRequest({
+        user: adminUser
+      })
+      call('get')
+    })
 
     it('should call the response method', () => {
-      expect(getServiceUsersStub.called).to.be.true // eslint-disable-line
-      expect(getInvitedUsersStub.called).to.be.true // eslint-disable-line
-      expect(responseStub.called).to.be.true // eslint-disable-line
+      expect(mockGetServiceUsers.called).to.be.true // eslint-disable-line
+      expect(mockResponse.called).to.be.true // eslint-disable-line
+      expect(mockResponse.called).to.be.true // eslint-disable-line
     })
 
     it('should pass req, res and template path to the response method', () => {
-      expect(responseStub.args[0]).to.include(req)
-      expect(responseStub.args[0]).to.include(res)
-      expect(responseStub.args[0]).to.include('simplified-account/settings/team-members/index')
+      expect(mockResponse.args[0][0].user).to.deep.include(adminUser)
```

Comment:
Don't we want to keep the testing of the `req` here? 
```
expect(mockResponse.args[0]).to.include(req)
```

Reply:
I think the use of `nextRequest` means that we won't expect req to remain the same. Looking at the other uses of the ControllerTestBuilder when they use `nextRequest`, they are only asserting on the part that we have added